osu.Game/Screens/OnlinePlay/Matchmaking/RankedPlay/DiscardScreen.cs

```diff
@@ -117,257 +106,87 @@ private void load()
-            }
-
-            delay += 1500;
+            Client.DiscardCards(playerHand.Selection.Select(it => it.Card).ToArray()).FireAndForget();
         }
 
-        private void flyInNewCards(ref double delay)
+        public override void CardAdded(RankedPlayScreen.Card card)
         {
-            if (cards.Count == 5)
-                return;
+            card.Position = new Vector2(DrawWidth, DrawHeight - 120);
+            card.Rotation = -30;
 
-            const double stagger = 100;
+            card.ChangeFacade(insertionFacade);
 
-            var newCards = new List<Card>();
+            double insertionTime = Math.Max(Time.Current, lastCardInsertionTime + 100);
+            lastCardInsertionTime = insertionTime;
 
-            while (cards.Count < 5)
-            {
-                var newCard = new Card(new MultiplayerPlaylistItem())
-                {
-                    Origin = Anchor.Centre,
-                    Anchor = Anchor.Centre,
-                    Alpha = 0,
-                    X = DrawWidth * 0.75f
-                };
+            double delay = insertionTime - Time.Current;
 
-                cards.Add(newCard);
-                cardFlow.Add(newCard);
-                newCards.Add(newCard);
-            }
+            card.FadeOut()
+                .Delay(delay)
+                .FadeIn();
 
-            var cardLayout = layoutCards(cards, -20)
-                             .Select(x => new { x.card, x.x, isNew = newCards.Contains(x.card) })
-                             .ToList();
-
-            foreach (var entry in cardLayout.Where(it => !it.isNew))
+            Scheduler.AddDelayed(() =>
             {
-                entry.card
-                     .Delay(delay)
-                     .MoveToX(entry.x, 1000, Easing.OutExpo)
-                     .MoveToY(cardYOffsetAt(entry.x), 1000, Easing.OutExpo)
-                     .RotateTo(cardRotationAt(entry.x), 1000, Easing.OutExpo);
+                var facade = playerHand.AddCard(card);
 
-                delay += stagger / 2;
-            }
+                card.ChangeFacade(facade);
 
-            foreach (var entry in cardLayout.Where(it => it.isNew))
-            {
-                entry.card
-                     .Delay(delay)
-                     .FadeIn()
-                     .MoveToX(entry.x, 600, Easing.OutExpo)
-                     .MoveToY(cardYOffsetAt(entry.x), 600, Easing.OutExpo)
-                     .RotateTo(cardRotationAt(entry.x), 600, Easing.OutExpo);
+                facade.CardMovementBindable.Value = RankedPlayScreen.MovementStyle.Smooth;
 
-                delay += stagger;
-            }
-
-            delay += 1000;
+                Scheduler.AddDelayed(() => facade.CardMovementBindable.Value = RankedPlayScreen.MovementStyle.Energetic, 300);
+            }, delay);
         }
 
-        private void presentRemainingCards(ref double delay)
+        public override void CardRemoved(RankedPlayScreen.Card card)
```

Comment:
Also here, I'm not sure if it was intentional that opponent's cards are moved to the centre (it's an interesting quirk). Though might be best to add the concept of the user to this method too?